File: src/main/java/it/contrader/dto/DocumentDTO.java
package it.contrader.dto;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import it.contrader.model.Description;
import it.contrader.model.Folder;
import it.contrader.model.User;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIdentityInfo(generator=ObjectIdGenerators.PropertyGenerator.class, property="id")
public class DocumentDTO {
	
	private Long id;
	
	private String text;
	
	private User user;
	
	private Folder folder;
	
	private Description description;


}

File: src/dto/documentdto.ts
import { UserDTO } from './userdto';
import { FolderDTO } from './folderdto';
import { DescriptionDTO } from './descriptiondto';

export class DocumentDTO {
    
    id: number;

    text: string;

    user : UserDTO;

    folder: FolderDTO;

    description: DescriptionDTO;
}